tests/test_requests.py

```diff
@@ -2138,3 +2138,18 @@ def test_preparing_bad_url(self, url):
         r = requests.Request('GET', url=url)
         with pytest.raises(requests.exceptions.InvalidURL):
             r.prepare()
+
+    @pytest.mark.parametrize(
+        'url',
+        (
+            b"http+unix://%2Fvar%2Frun%2Fsocket/path",
+            u"http+unix://%2Fvar%2Frun%2Fsocket/path",
+        )
+    )
+
+    def test_prepare_http_unix(self, url):
```

Comment:
So this is good, let's just make this a bit more tempting for people to extend in future by changing the name: `test_url_passthrough`.